### ISSUE TITLE
new team + gbxml validation

### ISSUE THREAD

**Issue by theo-armour** [ladybug-tools/spider#35]
@bwelle @antonszilasi @chiensiTB 

you are now all members of a new team: @ladybug-tools/build-well 

***

Am I correct in thinking that this validator is not that helpful at our current stage of development? Or do I just not understand how it works well enough?

Given the above, would it make sense for me to, biting the bullet, download and install Open Studio and so be able to verify Build Well gbXML export files in a more speedy fashion?

**Comment by bwelle** [comment]
I wouldn't use OS to validate. It may may break for non-validation reasons.

**Comment by bwelle** [comment]
Chien Si said he's not sure if it's how we are using it or if there's an issue with the validator.

**Comment by mdengusiak** [comment]
Yes, current validator is very confusing as only validates few test cases files. I would highly recommend this one created by Karlsruhe Institute of Technology.

https://www.iai.kit.edu/english/1302.php

Go to query and can see warnings/errors and save as xml file.
This could be aslo some inspiration for warning/error finding

![image](https://user-images.githubusercontent.com/14005893/31196872-1f2c12be-a947-11e7-8915-5cb2c0fd9544.png)

**Comment by bwelle** [comment]
Isn't this more of a viewer rather than validator? I.e it might be very forgiving in way we don't want it to be? Maybe not, just thinking out loud.

**Comment by mdengusiak** [comment]
Hi, it is viewer but also list all issues and errors and warning with gbXML structure. This could be good to see where file have fundamental problems.

**Comment by bwelle** [comment]
Got it, thanks.

**Comment by chiensiTB** [comment]
The validator that gbxml wrote is not intended to consume any gbxml file.  That is what the industry wants, but gbxml has yet to provide.  All the validator does is just allow you to draw a test case in your tool (spider/build well), export it, and upload it to the validator.

If you want something that is more generic, then the options are to import it into a desktop viewer/validator as some others suggested.  Ultimately, you'll want to put it through OS.

**Comment by bwelle** [comment]
I still say we want to validate visually or otherwise prior to OS. If it looks good, then we ping OS.

**Comment by bwelle** [comment]
Interestingly enough, I took the [gbXML file](url) that was one of the only test cases from Chien Si (clean files) that worked in OS, and I just uploaded and tested with whole building test and it failed miserably, couldn't even generate error log. Wonder what the other test numbers are. Am I using the right xsd?

**Comment by chiensiTB** [comment]
I don't think you guys are fully grasping what that validator is doing.

**Comment by bwelle** [comment]
Hmm, I assumed it checks for minimum required objects, correct surface definitions, adjacencies, coordinate ordering, etc. No?

**Comment by chiensiTB** [comment]
No.  Wrong assumption.

**Comment by bwelle** [comment]
Theo, I spoke with Chien Si, I get it now. The validator is looking for specific test cases, geometry, for vendor certification. It can't just just take any gbXML and generically check it. So it's useless to us. We should use the tool suggested above to inspect, or just pop it into the gbXML JSON viewer to check (I say we start with the former). If all looks good, give it to Anton and he will ping the OS API and see what happens. We go from there.

**Comment by mdengusiak** [comment]
There is few fundamental geometry checks that I would  be good to implement.
Currently many people use Revit to generate gbXML and code they use is not perfect. So would be good to have:
1. check that all internal surfaces have two **different** zones on both sides
   to deal with air gaps and ceilings
2. check for duplicate Element ID, GUID for all elements and list them if they are duplicate
3. double point
etc.
show all warning highlighted in Viewer when we click on warning, this would be awesome

**Comment by bwelle** [comment]
Hi Theo,

I can help with the validation. I did this stuff for thermal models for a long time (remember that document I sent you?!) So you can generate gbXML files, and I can visually inspect, and if looks good pass to Anton to ping the API with. Does this work for you? I'm assuming you want your own way to validate without a downloaded viewer.:) Maybe put into the gbXML JSON viewer or regular viewer I see on Spider?

**Comment by theo-armour** [comment]
Sounds good.

The dev version of BW gbxml exporter is now handling Box and L Shape
geometry. Now working adding the attributes. Should have something to
inspect soon. I am going to a friend's lecture on robotics in an hour.
Don't know when I will be back.

**Comment by theo-armour** [comment]
@mdengusiak 

Thanks for link to Karlsruhe Institute of Technology validator. I will download and try it out.

@ladybug-tools/build-well 

I have downloaded and installed Open Studio. I have started experimenting with it. I have been able to correct two issues it reported which then enabled me to progress further through the import process.

Currently Open Studio is reporting messages like this: Surface 'bw21' is not assigned to a space

Therefore it appears Open Studio is able to get all the way through the files Build Well > gbXML export  is creating. It appears that once the 'Space' items info are added, the import could be good. 

Fingers crossed, this gets dome in the next day or so.
